src/Toolkit/Date.php

```diff
@@ -70,20 +69,17 @@ public function ceil(string $unit)
 	/**
 	 * Returns the interval between the provided and the object's datetime
 	 *
-	 * @param string|int|\DateTimeInterface $datetime
-	 * @param \DateTimeZone|null $timezone Optional default timezone if `$datetime` is string
-	 * @return \DateInterval
+	 * @param \DateTimeZone $timezone Optional default timezone if `$datetime` is string
```

Comment:
This param is still optional (same a lot of times below). Why have you removed the `null` option?

```suggestion
	 * @param \DateTimeZone|null $timezone Optional default timezone if `$datetime` is string
```

Reply:
A similar situation happened here: https://github.com/getkirby/kirby/pull/5079#discussion_r1119076118. I think Nico removed it specifically for use cases where the `null` parameter is not sent.

Reply:
Don't we have that in many places where we mark parameters as optional via `= null` but don't add `null` again as type hint before the variable name? As this mainly should read "$timezone needs to be of DateTimeZone, but is optional", not that `null` is an actual type for `$timezone` that had another function/meaning.

Given, we also have a lot of places with still `|null` - but I think that's actually not right in most of these.

Reply:
@lukasbestle @afbora thoughts?

Reply:
But then it needs a `?` before the type, doesn't it? The docblock type overrides the type-hint, so for machines it's no longer visible that the argument is optional.

And given that we got rid of the `?` in a lot of places (or even everywhere?), I think that just leaves `|null`.

Reply:
Doesn't `= null` marks it for machine as well as optional? Isn't that the implication?

Reply:
Ok, now I understood it: you're fine with just `= null` in the actual PHP code, but in the DocBlock you think it's still required as this will take priority for the docs?

Reply:
Ah, I see what you mean. If you use `type $attr = null`, PHP automatically treats this as `type|null $attr = null` internally. But IMO this is a not an actual PHP feature but just a compatibility hack. They probably went with this auto-conversion to avoid that a lot of code bases break.

IMO we shouldn't rely on this compatibility hack and rather type-hint explicitly. An alternative could be the `?type` syntax (which to me looks like a good fit for the "optional argument" use case), but we already decided against that because it's not compatible with multiple types.

So I think that only leaves us `|null`.

Reply:
I added it. Also rebased the PR but only the last commit is new/has the new changes.